prow/github/helpers.go

```diff
@@ -82,3 +85,33 @@ func PermissionsFromLevel(permission RepoPermissionLevel) RepoPermissions {
 		return RepoPermissions{}
 	}
 }
+
+// This method is used to match content URL from the project card(cardURL) with the issue or pull request url(contentURL).
+// The contentURL can be a html_url or url field of the issue/pull request object.
+// ex: ContentURL is the object url of the issue/PR "url": "https://api.github.com/repos/ii/apisnoop/pulls/15"
+// where as the html_url is: https://github.com/ii/apisnoop/pull/15
+// Card always holds the object URL which is https://api.github.com/repos/ii/apisnoop/issues/15
+func MatchCardToContent(cardURL, contentURL string) bool {
+	// extract org, repo and issue number frmo cardURL,
+	// cardURL is well formed format, ex: https://api.github.com/repos/<org>/<repo>/issues/<issue num>
+	pattern := regexp.MustCompile("https://api.github.com/repos/(?P<org>\\w+)/(?P<repo>\\w+)/issues/(?P<num>\\d+)")
```

Comment:
One last comment -- if we have a method signature

```go
func (c *Client) GetColumnProjectCard(org, repo string, num, columndId int) (*github.ProjectCard, error)
```

We can then just call:

```
func MatchCardToContent(org, repo string, num int, cardURL string) bool
```

As the org, repo and pull number are all known values we should prefer to just pass them through instead of figuring them out once again here.

Reply:
GetColumnProjectCard is fetching the Card object. We did not get org, repo and number from it unless we process the contentURL or make an api call.
ex: card data
  {
    "url": "https://api.github.com/projects/columns/cards/20883500",
    "project_url": "https://api.github.com/projects/2501156",
    "id": 20883500,
    "node_id": "MDExOlByb2plY3RDYXJkMjA4ODM1MDA=",
    "note": null,
    "archived": false,
    "creator": {
      "login": "cncf-ci",
      "id": 43581769,
      "node_id": "MDQ6VXNlcjQzNTgxNzY5",
      "avatar_url": "https://avatars2.githubusercontent.com/u/43581769?v=4",
      "gravatar_id": "",
      "url": "https://api.github.com/users/cncf-ci",
      "html_url": "https://github.com/cncf-ci",
      "followers_url": "https://api.github.com/users/cncf-ci/followers",
      "following_url": "https://api.github.com/users/cncf-ci/following{/other_user}",
      "gists_url": "https://api.github.com/users/cncf-ci/gists{/gist_id}",
      "starred_url": "https://api.github.com/users/cncf-ci/starred{/owner}{/repo}",
      "subscriptions_url": "https://api.github.com/users/cncf-ci/subscriptions",
      "organizations_url": "https://api.github.com/users/cncf-ci/orgs",
      "repos_url": "https://api.github.com/users/cncf-ci/repos",
      "events_url": "https://api.github.com/users/cncf-ci/events{/privacy}",
      "received_events_url": "https://api.github.com/users/cncf-ci/received_events",
      "type": "User",
      "site_admin": false
    },
    "created_at": "2019-04-30T05:50:39Z",
    "updated_at": "2019-04-30T05:50:39Z",
    "column_url": "https://api.github.com/projects/columns/5090897",
    "content_url": "https://api.github.com/repos/ii/apisnoop/issues/13"
  }

Reply:
Right, but in `project.go` when you first make the call to `client.GetColumnProjectCard` you _do_ have that data.

Reply:
I have org and repo from the event and that is from the issue/PR not from project. Project may belong to a org only.

Also project may not belong to the same repo as the issue/pr, I am interested in matchingcard to an issue/pr and only source is the contentURL in the card. 
Sample of such case is using curl on conformance project ...
curl --user brahmaroutu:bc4ffe3b8da693d604218d677b3e8901adb61730   https://api.github.com/orgs/kubernetes/projects  -H "Accept:application/vnd.github.inertia-preview+json"
and then use the cards data (https://api.github.com/projects/2067392/columns) and get to one of the cards (https://api.github.com/projects/columns/4587456/cards) and there is a issue from the repo https://api.github.com/repos/kubernetes/community/issues/3361

Reply:
You're passing `e.IssueHTMLURL` into the `client.GetColumnProjectCard` call, which is using the issue number from the event. You are always looking for the card that matches the org/repo/issue number for the event you're handling. I'm just saying instead of passing in the HTML URL, pass in the parts.

Reply:
Project is searched under the repos first and if we don't find it then we go search the org(https://github.com/kubernetes/test-infra/blob/e17115ce9e99f6b604aa1ee75aca66e25e13c36a/prow/plugins/project/project.go#L202), example above is a project from the org(Kubernetes) and the issue is under (Kubernetes/community).

Reply:
But isn't the content of the card you're looking for the same every time?

Reply:
Yep @stevekuznetsov, you are right, may be I just always expect the url sent correctly by the originator. I changed code in the call to GetColumnProjectCard(in project.go) to pass the properly formed url. Please review.